src/core/tests/type_prop/reverse.cpp

```diff
@@ -155,39 +160,175 @@ TEST(type_prop, reverse_partial_rank_dynamic) {
+
+    OV_EXPECT_THROW(
+        auto op = make_op(param, axes, op::v1::Reverse::Mode::MASK),
+        NodeValidationFailure,
+        HasSubstr("The number of elements in the reversed_axes tensor (5) must match the input data tensor rank"));
+}
+
+TEST_F(TypePropReverseV1Test, axes_index_is_scalar) {
+    PartialShape param_shape{2, {2, 10}, 8};
+    auto param = make_shared<Parameter>(element::f32, param_shape);
+    auto axes = make_shared<Parameter>(element::i64, PartialShape{});
+
+    OV_EXPECT_THROW(auto op = make_op(param, axes, op::v1::Reverse::Mode::INDEX),
+                    NodeValidationFailure,
+                    HasSubstr("The reversed_axes input must be a 1D tensor"));
+}
+
+TEST_F(TypePropReverseV1Test, axes_mask_is_scalar) {
+    PartialShape param_shape{2, {2, 10}, 8};
+    auto param = make_shared<Parameter>(element::f32, param_shape);
+    auto axes = make_shared<Parameter>(element::boolean, PartialShape{});
+
+    OV_EXPECT_THROW(auto op = make_op(param, axes, op::v1::Reverse::Mode::MASK),
+                    NodeValidationFailure,
+                    HasSubstr("The reversed_axes input must be a 1D tensor"));
+}
+
+TEST_F(TypePropReverseV1Test, axes_mask_not_boolean_type) {
+    PartialShape param_shape{2, {2, 10}, 8};
+    auto param = make_shared<Parameter>(element::f32, param_shape);
+    auto axes = make_shared<Parameter>(element::i32, PartialShape{4});
+
+    OV_EXPECT_THROW(auto op = make_op(param, axes, op::v1::Reverse::Mode::MASK),
+                    NodeValidationFailure,
+                    HasSubstr("In 'mask' mode the second input must contain boolean values"));
+}
+
+TEST_F(TypePropReverseV1Test, axes_index_not_integer_type) {
+    PartialShape param_shape{2, {2, 10}, 8};
+    auto param = make_shared<Parameter>(element::f32, param_shape);
+    auto axes = make_shared<Parameter>(element::f32, PartialShape{4});
+
+    OV_EXPECT_THROW(auto op = make_op(param, axes, op::v1::Reverse::Mode::INDEX),
+                    NodeValidationFailure,
+                    HasSubstr("In 'index' mode the second input must contain integer values"));
+}
+
+TEST_F(TypePropReverseV1Test, param_static_rank_partial_shape_axes_out_of_input_rank) {
+    PartialShape param_shape{Dimension::dynamic(), Dimension::dynamic(), 2, 3};
+    auto param = make_shared<op::Parameter>(element::f32, param_shape);
+
+    OV_EXPECT_THROW(
+        auto op = make_op(param, Constant::create(element::i64, {3}, {0, 4, 2}), op::v1::Reverse::Mode::INDEX),
+        NodeValidationFailure,
+        HasSubstr("Some of the provided axes (AxisSet{0, 2, 4}) are out of bounds (input rank: 4)."));
+}
+
+TEST_F(TypePropReverseV1Test, param_static_rank_partial_shape_axes_negatives) {
+    PartialShape param_shape{-1, {2, -1}, {-1, 3}, 5};
+    set_shape_labels(param_shape, 10);
+    auto param = make_shared<op::Parameter>(element::f32, param_shape);
+
+    auto op = make_op(param, Constant::create(element::i64, {3}, {0, -1, 2}), op::v1::Reverse::Mode::INDEX);
```

Comment:
Should negative axis values be allowed and pass the validation?

Unfortunately, there is no direct info about negative axes support in the [Reverse_1.md](https://github.com/openvinotoolkit/openvino/blob/da9470864c1e099f46611e1db9534ad0c89d4b24/docs/ops/movement/Reverse_1.md) operation specification.

Reply:
Yes, I've got same doubts but I left this behavior unchanged.
It was hidden in `get_axis_set_val` when get AxisSet from constant value.